plugins/synapticsmst/fu-plugin-synapticsmst.c

```diff
@@ -15,6 +15,30 @@
 #define SYNAPTICS_FLASH_MODE_DELAY 3
 #define SYNAPTICS_UPDATE_ENUMERATE_TRIES 3
 
+static gboolean
+syanpticsmst_check_amdgpu_safe (GError **error)
```

Comment:
do you think we should factor this out so that we have something like `gboolean fu_common_has_kernel_module(const gchar *modname)` -- this seems like the kind of thing that could be used elsewhere, e.g. wacom.

Reply:
My only hesitation is that this doesn't cover the situation of "built-in to the kernel" so it's not generalized enough to test a _success_ case (Do I have a module needed to support something; abort plugin if not).

Reply:
I didn't think about the _built-in_ case. Is there any way at all to detect that?

Reply:
I don't think so, it's just treated like core kernel code then isn't it?  So instead you'd have to look for a signature of the driver (like a sysfs directory or node being populated).

Reply:
Sure, would it be possible for look for something in /sys that would detect both cases?

Reply:
If we go back to finding something in `/sys` this turns into `g_file_test` for something existing rather than a general helper function.  I don't know what the right thing to check for in sysfs with amdgpu is though.

Reply:
Is `amdgpu` likely to be built into the kernel? If the answer is "basically never" then I think this is fine, if it's anything more than "never" then I think we need to do a local g_file_test thing instead.

Reply:
The answer is basically never.  You would only build it into the kernel if you're doing a super specific kernel build for your machine.  None of the distro kernels will ever do this.